engine/dlib/src/dlib/mutex_posix.cpp

```diff
@@ -14,9 +14,16 @@
 
 #include <assert.h>
 #include "mutex.h"
+#include <dmsdk/dlib/log.h>
 
 namespace dmMutex
 {
+    #define CHECK_RET(ret) \
+        if (ret) { \
+            dmLogError("%s:%d failed: %d", __FUNCTION__, __LINE__, ret); \
+            assert(ret == 0); \
+        }
```

Comment:
Better to add this diagnostic for any future threading issues, to print out the actual return values.

This time I got 16, which was EBUSY on my platform, and it led be to the actual problem.